src/syscall_handler_window.c

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright (C) 2016 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+#include "syscall_handler_window.h"
+#include "process.h"
+
+#define CHECK_PROC_WINDOW() if(current->window == 0) return -1
+
+int32_t sys_window_create(int x, int y, int width, int height) {
+    CHECK_PROC_WINDOW();
```

Comment:
Why is this "called" here? You step into sys_window_create() and the first thing it seems to do is return -1 if current->window is null. Is this the intended behavior?